tests/std/tests/P0355R7_calendars_and_time_zones_hms/test.cpp

```diff
@@ -0,0 +1,174 @@
+    static_assert(width<1, 7>() == 6);
+    static_assert(width<1, 8>() == 3);
+    static_assert(width<1, 9>() == 6);
+    static_assert(width<1, 10>() == 1);
+    static_assert(width<756, 625>() == 4);
+    static_assert(width<3780, 625>() == 3);
+}
+
+constexpr void constructor() {
+    static_assert(noexcept(hms_hours{}));
+
+    assert(hms_hours{}.hours() == hms_hours{hours::zero()}.hours());
+    assert(hms_hours{}.minutes() == hms_hours{hours::zero()}.minutes());
+    assert(hms_hours{}.seconds() == hms_hours{hours::zero()}.seconds());
+    assert(hms_hours{}.subseconds() == hms_hours{hours::zero()}.subseconds());
+}
+
+constexpr void is_negative() {
+    static_assert(noexcept(hms_hours{}.is_negative()));
+
+    assert(hh_mm_ss<days>(days{-1}).is_negative());
+    assert(!hh_mm_ss<days>(days{1}).is_negative());
+
+    assert(hh_mm_ss<hours>(-1h).is_negative());
+    assert(!hh_mm_ss<hours>(1h).is_negative());
+
+    assert(hh_mm_ss<minutes>(-1min).is_negative());
+    assert(!hh_mm_ss<minutes>(1min).is_negative());
+
+    assert(hh_mm_ss<seconds>(-1s).is_negative());
+    assert(!hh_mm_ss<seconds>(1s).is_negative());
+
+    assert(hh_mm_ss<milliseconds>(-1ms).is_negative());
+    assert(!hh_mm_ss<milliseconds>(1ms).is_negative());
+
+    assert(hh_mm_ss<microseconds>(-1us).is_negative());
+    assert(!hh_mm_ss<microseconds>(1us).is_negative());
+
+    assert(hh_mm_ss<nanoseconds>(-1ns).is_negative());
+    assert(!hh_mm_ss<nanoseconds>(1ns).is_negative());
+}
+
+constexpr auto ones = 1h + 1min + 1s + 1ms;
+
+constexpr void hour() {
+    static_assert(noexcept(hms_hours{}.hours()));
+
+    assert(hh_mm_ss(days{1}).hours() == 24h);
+    assert(hh_mm_ss(ones).hours() == 1h);
+    assert(hh_mm_ss(-ones).hours() == 1h);
+    assert(hh_mm_ss(59min).hours() == 0h);
+}
+
+constexpr void mins() {
+    static_assert(noexcept(hms_hours{}.minutes()));
+
+    assert(hh_mm_ss(ones).minutes() == 1min);
+    assert(hh_mm_ss(-ones).minutes() == 1min);
+    assert(hh_mm_ss(59s).minutes() == 0min);
+}
+
+constexpr void secs() {
+    static_assert(noexcept(hms_hours{}.seconds()));
+
+    assert(hh_mm_ss(ones).seconds() == 1s);
+    assert(hh_mm_ss(-ones).seconds() == 1s);
+    assert(hh_mm_ss(999ms).seconds() == 0s);
+}
+
+constexpr void subsecs() {
+    static_assert(noexcept(hms_hours{}.subseconds()));
+
+    assert(hh_mm_ss(ones).subseconds() == 1ms);
+    assert(hh_mm_ss(-ones).subseconds() == 1ms);
+    assert(hh_mm_ss(999us).subseconds() == 999us);
+    assert(hh_mm_ss(duration_cast<milliseconds>(999us)).subseconds() == 0ms);
+}
+
+constexpr void to_duration() {
+    using precision = hms_hours::precision;
+
+    static_assert(noexcept(hms_hours{}.to_duration()));
+    static_assert(noexcept(static_cast<precision>(hms_hours{})));
+
+    assert(hh_mm_ss(ones).to_duration() == ones);
+    assert(hh_mm_ss(-ones).to_duration() == -ones);
+
+    hh_mm_ss<milliseconds> hms(50ms);
+    milliseconds milli = static_cast<milliseconds>(hms);

```

Comment:
This shadows `std::milli`, a `ratio` typedef (which is, in practice, dragged in by `<chrono>`, and this test is `using namespace std;`). Using any other identifier, like `milli_val`, would avoid this.